### PR TITLE
[Integration manager] register dw connector

### PR DESCRIPTION
**Pull request by jnurse**
This PR adds registration functionality to the demandware connector. 

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1258
 **Linked PRs**: n/a

## Changes
- Added command to register user and login 

## How to test-drive this PR
- Run `npm run dev`
- Preview [the demandware site](https://preview.mobify.com/?url=http%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Open the site menu and click "sign in"
- Click "Register"
- Enter valid details for a user that hasn't been created yet
- The user is created without any errors.

### CODE REVIEW

File: web/app/integration-manager/_demandware-connector/login/commands.js [hunk 1]
```diff
@@ -75,6 +75,27 @@ export const login = ({login}) => (dispatch) => {
         })
 }
 
-export const registerUser = () => (dispatch) => {
-    return dispatch()
+export const registerUser = (href, {firstname, lastname, email, password}, resolve, reject) => (dispatch) => {
```
Comment (DanielBrookRoberge) [inline web/app/integration-manager/_demandware-connector/login/commands.js]: This doesn't match the way this function is called in `login/actions.js` now after the refactoring on the sign in PR.

File: web/app/integration-manager/_demandware-connector/login/commands.js [hunk 2]
```diff
@@ -75,6 +75,27 @@ export const login = ({login}) => (dispatch) => {
         })
 }
 
-export const registerUser = () => (dispatch) => {
-    return dispatch()
+export const registerUser = ({firstname, lastname, email, password}, resolve, reject) => (dispatch) => {
```
Comment (DanielBrookRoberge) [inline web/app/integration-manager/_demandware-connector/login/commands.js]: Are we still passing in `resolve` and `reject`?

File: web/app/integration-manager/_demandware-connector/login/commands.js [hunk 3]
```diff
@@ -71,10 +71,30 @@ export const login = ({login}) => (dispatch) => {
...
+                login: email,
+                email
+            }
+        })
+    }
+    return makeDemandwareRequest(`${API_END_POINT_URL}/customers`, requestOptions)
+        .then((response) => response.json())
+        .then((responseJSON) => {
+            if (responseJSON.fault) {
+                throw new SubmissionError({_error: 'Unable to create account.'})
```
Comment (jeremywiebe) [inline web/app/integration-manager/_demandware-connector/login/commands.js]: It seems like we aren't handling this properly in `containers/login/actions.js`.  We do `.catch()` in that action, but we call the `reject()` method in that catch which doesn't seem to do anything. So if the server rejects the call or fails, the UI doesn't show any errors.

Comment (jeremywiebe) [inline web/app/integration-manager/_demandware-connector/login/commands.js]: It's working properly. :( I just didn't see the error at the top of the form.

**Pull request merged**